packages/integration-tests/src/utils.rs

```diff
@@ -176,8 +177,15 @@ pub fn proper_instantiate() -> (App, CwTemplateContract) {
 }
 
 pub fn get_dr_id(res: AppResponse) -> String {
-    // TODO: this is an ugly way to get the dr_id.
-    // although PostDataRequest on the DataRequest contract returns it in `data`, the Proxy contract does not yet.
-    // https://github.com/sedaprotocol/seda-chain-contracts/issues/68
-    res.events.last().unwrap().attributes[2].value.clone()
+    let binary = res.data.unwrap();
+
+    // I don't know why protobuf adds 2 bytes to the beginning of the data payload but it does
+    let binary = &binary.to_vec()[2..];
+
+    let dr_id = String::from_utf8(binary.to_vec()).unwrap();
+
+    // remove first and last char (they are quotes)
+    let dr_id = &dr_id[1..dr_id.len() - 1];
```

Comment:
I imagine this is JSON serialisation so maybe we can just deserialise the bytes directly?

Reply:
Also from my previous comment, I believe that the quotes are because what Is how with my little test (more specifically `t1`).

```
recovered text 1: "3a6eb0790f39ac87c94f3856b2dd2c5d110e6811602261a9a923d3bb23adc8b7"
```